ush/ocnpost.py

```diff
@@ -0,0 +1,646 @@
+        sinrot2d[0,:,:]=sinrot[:]
+        cosrot2d[0,:,:]=cosrot[:]
+
+        print('Conforming rotation arrays to 3d')
+        sinrot3d=np.zeros(shape=(mask3d.shape),dtype=np.float64)
+        cosrot3d=np.zeros(shape=(mask3d.shape),dtype=np.float64)
+        for i in range(0,nlevs):
+            sinrot3d[0,i,:,:]=sinrot[:]
+            cosrot3d[0,i,:,:]=cosrot[:]
+
+    if model == 'CICE': 
+        mask2d = np.where(np.isnan(SST),  1.0, 0.0)
+        angleT2d=np.zeros(shape=(mask2d.shape),dtype=np.float64)
+        angleT2d[0,:,:]=angleT[:]
+
+    print('Checking which scalar variables are available in input file')
+    #check for scalar variables in file. this is only required because
+    #of the missing/misnamed MLD variables in the first BM
+    #only the varlist is checked, since it is assumed there are
+    #no other variables missing after the first benchmark
+    valid=np.zeros(shape=(nvars),dtype=bool)
+    valid[:]=False
+    for nv in range(0,nvars):
+        varname = varlist[nv,0]
+        if varname in ncvarlist:
+            valid[nv]=True
+        print(varlist[nv,0]+'   '+str(valid[nv]))
+
+    #check for vector variables in file. 
+    #If at least one component of the vector is not in input file, 
+    #then the entire vector pair will be ignorned
+    validv=np.zeros(shape=(nvpairs),dtype=bool)
+    validv[:]=False
+    for nv in range(0,nvpairs):
+        vecnames = veclist[nv,0,:]
+        for nv2 in range(0,2):
+            vecname=vecnames[nv2]
+            if vecname in ncvarlist:
+                validv[nv]=True
+            else:
+                validv[nv]=False
+                break
+        print(veclist[nv,0,0]+' and '+veclist[nv,0,1]+'  '+str(validv[nv]))
+
+    #----------------------------------------------------------------------
+    # loop over the output resolutions 
+    #----------------------------------------------------------------------
+
+    for jj in range(0,len(dstgrds)):
+
+        outres = dstgrds[jj]+'x'+dstgrds[jj]
+        outgrid = dstgrds[jj]
+
+        if outgrid == '0p25':
+            lond = np.arange(0, 360.0, 0.25)
+            latd = np.arange(-90.0, 90.25, 0.25)
+        if outgrid == '0p5':
+            lond = np.arange(0, 360.0, 0.5)
+            latd = np.arange(-90.0, 90.5, 0.5)
+        if outgrid == '1p0':
+            lond = np.arange(0, 360.0, 1.0)
+            latd = np.arange(-90.0, 90.1, 1.0)
+
+        wgtsfile1 = nemsrc+'/'+'tripole.mx025.Ct.to.'+dsttype[0]+dstgrds[jj]+'.bilinear.nc'        
+        wgtsfile2 = nemsrc+'/'+'tripole.mx025.Ct.to.'+dsttype[0]+dstgrds[jj]+'.conserve.nc'
+        if model == 'MOM':
+            wgtsfile3 = nemsrc+'/'+'tripole.mx025.Cu.to.Ct.bilinear.nc'    
+            wgtsfile4 = nemsrc+'/'+'tripole.mx025.Cv.to.Ct.bilinear.nc'
+        if model == 'CICE':
+            wgtsfile3 = nemsrc+'/'+'tripole.mx025.Bu.to.Ct.bilinear.nc'    
+            wgtsfile4 = nemsrc+'/'+'tripole.mx025.Bu.to.Ct.bilinear.nc'
```

Comment:
these seem to be hard-wired here. Why?

Reply:
Yes, I will fix this so that the filenames not hard-wired. Thanks for catching this.

Reply:
All these things that are different for MOM and CICE should ideally be moved to a yaml file. The python shouldn't need to know what it is working on, just a yaml with all the settings.

Reply:
Agreed. yaml files will be used instead of the preliminary parm files that are being used now.